src/number.rs

```diff
@@ -1,29 +1,33 @@
 use std::ops::Deref;
 
+use crate::isolate::Isolate;
 use crate::support::Opaque;
-use crate::Isolate;
+use crate::value::Value;
+use crate::HandleScope;
 use crate::Local;
-use crate::Value;
 
 extern "C" {
-  fn v8__Number__New(isolate: &Isolate, value: f64) -> *mut Number;
+  fn v8__Number__New(isolate: *mut Isolate, value: f64) -> *mut Number;
   fn v8__Number__Value(this: &Number) -> f64;
-  fn v8__Integer__New(isolate: &Isolate, value: i32) -> *mut Integer;
+  fn v8__Integer__New(isolate: *mut Isolate, value: i32) -> *mut Integer;
   fn v8__Integer__NewFromUnsigned(
-    isolate: &Isolate,
+    isolate: *mut Isolate,
     value: u32,
   ) -> *mut Integer;
-  fn v8__Integer__Value(this: &Integer) -> i64;
+  fn v8__Integer__Value(this: *const Integer) -> i64;
 }
 
 /// A JavaScript number value (ECMA-262, 4.3.20)
 #[repr(C)]
 pub struct Number(Opaque);
 
 impl Number {
-  pub fn new(isolate: &Isolate, value: f64) -> Local<Number> {
+  pub fn new<'sc>(
+    scope: &mut HandleScope<'sc>,
+    value: f64,
+  ) -> Local<'sc, Number> {
```

Comment:
Doesn't match V8's API

Reply:
Because in v8 the handlescope is implicit to the active isolate. Here it's explicit because we want to know it's lifetime. What's the problem?